content/_components/onedrive/index.md

```diff
@@ -6,152 +6,158 @@ description: This is the component for working with Microsoft OneDrive storage s
 
-Action to delete item from OneDrive by provided path in selected disc.
-Returns filename if file was deleted and empty message if already wasn't exist
+* **Path** - (string, required): Full path to item, ex: `Monthly reports/November/Cars sales.pdf`
 
-![Delete File](img/delete-file.png)
+### Get File
 
-#### Input fields description
+Lookup a single file by its path.
 
-* **Drive Identity** - OneDrive instance to work with. Selects by owner
+#### Configuration Fields
 
-#### Metadata fields description
+* **Drive Identity** - (dropdown, required): OneDrive instance to work with
+* **Enable File Attachments** - (checkbox, optional): If checked, file will be uploaded to local storage and link provided in response
 
-* **Path** - Full path to item to delete
+#### Input Metadata
 
-#### Input example:
+* **Path** - (string, required): Full path to item, ex: `Monthly reports/November/Cars sales.pdf`
 
-```
-{
-    "path": "base_folder/inner_folder/file.any"
-}
-```
+#### Output Metadata
 
-### Create Folder
+File information as JSON object, if `Enable File Attachments` checked, there also will be additional field `attachmentUrl` with link to file on platform
 
-Create new folder in provided `path`. If `path` not exist component will fail.
+### Upsert File
 
-![Create Folder](img/create-folder.png)
+Updates (if record exist) or creates a new file
 
-#### Input fields description
+#### Configuration Fields
 
-* **Drive Identity** - OneDrive instance to work with. Selects by owner
-* **Conflict Behaviour** - behaviour in case folder already exists. Default: `Fail`. Options: `Fail`, `Replace`, `Rename`.
-    1. `Fail` - fails if folder with same name already exists under provided `path`
-    2. `Rename` - rename folder if folder with same name already exists under provided `path`. Examples: `exists` -> `exists_1`, `exists_1` -> `exists_1_1`
-    3. `Replace` - replace folder if folder with same name already exists under provided `path`. Note: files inside folder will not be replaced, but last modified date of folder will be updated
+* **Drive Identity** - (dropdown, required): OneDrive instance to work with
+* **Upload single file** - (checkbox, optional): Use this option if you want to upload a single file
 
-#### Metadata fields description
+#### Input Metadata
+If `Upload single file` checked, there will be 2 fields:
+* **URL** - (string, required): link to file on Internet or platform
+* **Path** - (string, required): Full path to item on OneDrive, ex: `Monthly reports/November/Cars sales.pdf`
 
-* **Path** - Path to to folder where new folder will be created. Use empty string or `/` for root
-* **Name** - Name of new folder
+If `Upload single file` unchecked:
+* **Files** - (array, required): Collection of files to upload, each record contains object with two keys:
+  * **URL** - (string, required): link to file on Internet or platform
+  * **Path** - (string, required): Full path to item on OneDrive, ex: `Monthly reports/November/Cars sales.pdf`
```

Comment:
* **Path** - (string, required): Full path to an item on OneDrive, ex: `Monthly reports/November/Cars sales.pdf`